File: src/main/java/com/golubovich/library/spring/model/Person.java
package com.golubovich.library.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "person")
public class Person {
    @Id
    @GeneratedValue
    private long id;
    private String email;
    private String password;

    private Role role;

    public Person(String email, String password) {
        this.email = email;
        this.password = password;
        this.role = Role.USER;
    }


    // чтобы @Data сгенерировала все, кроме сеттера id
    private void setId(long id) {
        this.id = id;
    }
}

File: src/app/models/person.ts
export class Person {
  id!: number;
  email!: string;
  password!: string;
  role!: Role;

  constructor(email: string, password: string, role: Role) {
    this.email = email;
    this.password = password;
    this.role = role;
  }
}


export enum Role {
  USER='USER',
  ADMIN='ADMIN'
}
